File: C++/프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> operations) {
    vector<int> q;
    string command, number;
    for(auto& o : operations) {
        number = string(o.begin() + 2, o.end());
        if(o[0] == 'I') { // Insert Number
            q.push_back(stoi(number));
        } else if (o[0] == 'D' && !q.empty()) { // Delete Number
            sort(q.begin(), q.end());
            q.erase(number == "-1" ? q.begin() : q.end() - 1);
        }
    }
    
    if(q.empty()) q.push_back(0);
    return {*max_element(q.begin(),q.end()), *min_element(q.begin(),q.end())};
}

File: Python3/프로그래머스/3/42628. 이중우선순위큐/이중우선순위큐.py
def solution(operations):
    order, number = None, None
    queue = []
    for o in operations:
        order, number = o.split(" ")
        if order == "I": #삽입 연산
            queue.append(int(number))
        elif order == "D" and len(queue) > 0: #삭제 연산
            queue.sort()
            if number == "-1":
                del queue[0]
            else:
                del queue[-1]
    
    if len(queue) <= 0:
        queue.append(0)
    return [max(queue), min(queue)]